investment-app/TRINV.Api/TRINV.Application/Commands/NewsComment/UpdateNewsCommentCommand.cs

```diff
@@ -0,0 +1,55 @@
+﻿namespace TRINV.Application.Commands.NewsComment;
+
+using System.ComponentModel.DataAnnotations;
+using Domain.Entities;
+using Interfaces;
+using MediatR;
+using Shared.Business.Exceptions;
+using Shared.Business.Extension;
+using Shared.Business.Utilities;
+
+public record UpdateNewsCommentCommand : IRequest<OperationResult>
+{
+    [Required]
+    public int NewsCommentId { get; set; }
+
+    [Required]
+    [MaxLength(200)]
+    public string Comment { get; set; } = string.Empty;
+}
+
+internal class UpdateNewsCommentCommandHandler : IRequestHandler<UpdateNewsCommentCommand, OperationResult>
+{
+    readonly IUnitOfWork _unitOfWork;
+    readonly IRepository<NewsComment> _newsCommentRepository;
+    readonly IUserContext _userContext;
+
+    public UpdateNewsCommentCommandHandler(
+        IUnitOfWork unitOfWork,
+        IRepository<NewsComment> newsCommentRepository, 
+        IUserContext userContext)
+    {
+        _unitOfWork = unitOfWork;
+        _newsCommentRepository = newsCommentRepository;
+        _userContext = userContext;
+    }
+
+    public async Task<OperationResult> Handle(UpdateNewsCommentCommand request, CancellationToken cancellationToken)
+    {
+        var operationResult = new OperationResult();
+
+        var newsComment = await _newsCommentRepository.GetByIdAsync(request.NewsCommentId, cancellationToken);
+
+        if (newsComment == null || newsComment.NewsId != _userContext.UserId)
+            return operationResult.ReturnWithErrorMessage(
+                new NotFoundException("Comment not found"));
```

Comment:
Don't throw different types of messages. Need to be consistant.
For example you start with {object} with id {id} was not found. Continue with that.

Maybe move it in to constant

Reply:
Done.